src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -135,7 +135,7 @@ class TreeChecker extends Phase with SymTransformer {
     }
   }
 
-  class Checker(phasesToCheck: Seq[Phase]) extends ReTyper {
+  class Checker(phasesToCheck: Seq[Phase]) extends ReTyper with NoChecking {
```

Comment:
That's a drastic change. Why disable all checking in TreeChecker?

If it's just checkDerivedValue class that should be disabled during tree checking then it's better to 
override that method directly.